src/port_forwarding.rs

```diff
@@ -0,0 +1,67 @@
+use core::fmt;
+use core::num::NonZeroU32;
+
+#[cfg(feature = "mux_client")]
+use super::mux_client_impl;
+
+/// Type of forwarding
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Hash)]
+pub enum ForwardType {
+    /// Forward requests to a port on the local machine to remote machine.
+    Local,
+
+    /// Forward requests to a port on the remote machine to local machine.
+    Remote,
+}
+
+#[cfg(feature = "mux_client")]
+impl From<ForwardType> for mux_client_impl::ForwardType {
+    fn from(fwd_type: ForwardType) -> Self {
+        use mux_client_impl::ForwardType::*;
+
+        match fwd_type {
+            ForwardType::Local => Local,
+            ForwardType::Remote => Remote,
+        }
+    }
+}
+
+/// TCP/Unix socket
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Hash)]
+pub enum Socket<'a> {
+    /// Unix socket.
+    UnixSocket {
+        /// Filesystem path
+        path: &'a str,
+    },
+
+    /// Tcp socket.
+    TcpSocket {
```

Comment:
I think this should just hold a `std::net::SocketAddr`, and then we should have a constructor for it that takes `impl std::net:::ToSocketAddrs`.

Reply:
Done, however I cannot `impl<T: std::net:::ToSocketAddrs> TryFrom<T> for Socket` due to conflict, unless I create another new type.

Reply:
Oh, I wasn't imagining a `TryFrom` — just have that be `fn new` instead.

Reply:
OK, I've implemented `Socket::new` to convert any `ToSocketAddrs` to `Socket`.